src/client/Client.cc

```diff
@@ -12214,8 +12262,19 @@ int Client::_getxattr(Inode *in, const char *name, void *value, size_t size,
 		      const UserPerm& perms)
 {
   int r;
+  const VXattr *vxattr = nullptr;
+
+  // for new getvxattr RPC
+  if (strncmp(name, "ceph.dir.layout", 15) == 0 ||
```

Comment:
@vshankar and @batrick
How do we deal with crashing servers where an op is not valid ?
We'd need to make the change in quincy to avoid crashing and return `-ENOSYS` for future versions of clients with newer ops.

So, this specific case is a bit hairy.

Reply:
@vshankar To answer your specific question: this change was supposed to pass on the xattr name in a transparent manner to the server and blindly return whatever the server responded with to the user app. However, @batrick reminded me to take care of the case where new client connected to an old server and issued the `getvxattr` op.
The new `getvxattr` op, supports only layout and pinning xattrs and hence the need to add this check.
How many other xattrs we'd need to eventually support is unknown at this time.

Reply:
> How do we deal with crashing servers where an op is not valid ? We'd need to make the change in quincy to avoid crashing and return `-ENOSYS` for future versions of clients with newer ops.
> 
> So, this specific case is a bit hairy.

Doesn't the server respond with ENOTSUPP for invalid client ops? You mention about the crash due to the server collecting op latency stats for getvxattr, however, the server should have responded (with ENOTSUPP) well before, isn't it?

Reply:
my bad ... gross oversight on my part to miss checking for `EOPNOTSUPP`

Reply:
now checking for feature `SERVER_QUINCY` before forwarding `getvxattr` to server
checking for `EOPNOTSUPP` return value after forwarding call to server does not help; servers crash in this case while attempting to collect op latency for unknown ops